src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -264,8 +272,27 @@ float64_t CCrossValidation::evaluate_one_run()
 		m_machine->set_store_model_features(true);
 
 		/* do actual cross-validation */
+#pragma omp parallel for
 		for (index_t i=0; i <num_subsets; ++i)
 		{
+			CMachine* machine;
```

Comment:
OK
I suggest for now: `CSGObject::clone`, which gives a deep copy.
Try that and see how bad it is ;)